diff --git a/node_modules/react-native-receive-sharing-intent/android/gradle.properties b/node_modules/react-native-receive-sharing-intent/android/gradle.properties
index 9303aa4..97c04bb 100644
--- a/node_modules/react-native-receive-sharing-intent/android/gradle.properties
+++ b/node_modules/react-native-receive-sharing-intent/android/gradle.properties
@@ -1,4 +1,4 @@
-ReceiveSharingIntent_kotlinVersion=1.3.50
+ReceiveSharingIntent_kotlinVersion=1.5.20
 ReceiveSharingIntent_compileSdkVersion=29
 ReceiveSharingIntent_buildToolsVersion=29.0.2
 ReceiveSharingIntent_targetSdkVersion=29
diff --git a/node_modules/react-native-receive-sharing-intent/android/src/main/java/com/reactnativereceivesharingintent/ReceiveSharingIntentHelper.java b/node_modules/react-native-receive-sharing-intent/android/src/main/java/com/reactnativereceivesharingintent/ReceiveSharingIntentHelper.java
index 69126eb..2ffe32e 100644
--- a/node_modules/react-native-receive-sharing-intent/android/src/main/java/com/reactnativereceivesharingintent/ReceiveSharingIntentHelper.java
+++ b/node_modules/react-native-receive-sharing-intent/android/src/main/java/com/reactnativereceivesharingintent/ReceiveSharingIntentHelper.java
@@ -27,15 +27,16 @@ public class ReceiveSharingIntentHelper {
   }
 
   @RequiresApi(api = Build.VERSION_CODES.KITKAT)
-  public void sendFileNames(Context context, Intent intent, Promise promise){
+  public boolean sendFileNames(Context context, Intent intent, Promise promise){
     try {
       String action = intent.getAction();
       String type = intent.getType();
-      if(type == null) { return; }
+      if(type == null) { return false; }
       if(!type.startsWith("text") && (Objects.equals(action, Intent.ACTION_SEND) || Objects.equals(action, Intent.ACTION_SEND_MULTIPLE))){
         WritableMap files = getMediaUris(intent,context);
-        if(files == null) return;
+        if(files == null) return false;
         promise.resolve(files);
+        return true;
       }else if(type.startsWith("text") && Objects.equals(action, Intent.ACTION_SEND)){
         String text = null;
         String subject = null;
@@ -45,8 +46,9 @@ public class ReceiveSharingIntentHelper {
         }catch (Exception ignored){ }
         if(text == null){
           WritableMap files = getMediaUris(intent,context);
-          if(files == null) return;
+          if(files == null) return true;
           promise.resolve(files);
+          return true;
         }else{
           WritableMap files = new WritableNativeMap();
           WritableMap file = new WritableNativeMap();
@@ -64,6 +66,7 @@ public class ReceiveSharingIntentHelper {
           file.putString("subject", subject);
           files.putMap("0",file);
           promise.resolve(files);
+          return false;
         }
 
       }else if(Objects.equals(action, Intent.ACTION_VIEW)){
@@ -79,6 +82,7 @@ public class ReceiveSharingIntentHelper {
         file.putString("extension", null);
         files.putMap("0",file);
         promise.resolve(files);
+        return true;
       }
       else if (Objects.equals(action, "android.intent.action.PROCESS_TEXT")) {
         String text = null;
@@ -96,11 +100,14 @@ public class ReceiveSharingIntentHelper {
           file.putString("text", text);
           files.putMap("0", file);
           promise.resolve(files);
+          return true;
       }else{
         promise.reject("error","Invalid file type.");
+        return false;
       }
     }catch (Exception e){
       promise.reject("error",e.toString());
+      return false;
     }
   };
 
diff --git a/node_modules/react-native-receive-sharing-intent/android/src/main/java/com/reactnativereceivesharingintent/ReceiveSharingIntentModule.java b/node_modules/react-native-receive-sharing-intent/android/src/main/java/com/reactnativereceivesharingintent/ReceiveSharingIntentModule.java
index f752144..a7d05b7 100644
--- a/node_modules/react-native-receive-sharing-intent/android/src/main/java/com/reactnativereceivesharingintent/ReceiveSharingIntentModule.java
+++ b/node_modules/react-native-receive-sharing-intent/android/src/main/java/com/reactnativereceivesharingintent/ReceiveSharingIntentModule.java
@@ -39,8 +39,9 @@ public class ReceiveSharingIntentModule extends ReactContextBaseJavaModule {
     Activity mActivity = getCurrentActivity();
     if(mActivity == null) { return; }
     Intent intent = mActivity.getIntent();
-    receiveSharingIntentHelper.sendFileNames(reactContext, intent, promise);
-    mActivity.setIntent(null);
+    if (receiveSharingIntentHelper.sendFileNames(reactContext, intent, promise)){
+      mActivity.setIntent(null);
+    }
   }
 
   @ReactMethod
